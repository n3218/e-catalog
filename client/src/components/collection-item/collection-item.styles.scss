.collection-item {
    // width: 22vw;
    display: flex;
    flex-direction: column;
    height: 350px;
    align-items: center;
    position: relative;
    margin-bottom: 30px;

    .image {
        width: 100%;
        height: 95%;
        background-size: cover;
        background-position: center;
        margin-bottom: 5px;
    }

    button {
        width: 80%;
        opacity: 0.7;
        position: absolute;
        top: 255px;
        display: none;
        @media screen and (max-width: 800px) {
            display: block;
            opacity: 0.9;
            min-width: unset;
            padding: 0 10px;
        }
    }

    &:hover {
        .image {
            opacity: 0.8;
        }
        button {
            opacity: 0.85;
            display: flex;
        }
    }

    .item-footer {
        width: 100%;
        height: 5%;
        display: flex;
        justify-content: space-between;
        font-size: 18px;

        .name {
            width: 90%;
            margin-bottom: 15px;
        }

    }
    @media screen and (max-width: 800px) {
        width: 100%;
        &:hover {
            .image {
                opacity: unset;
            }
            button {
                opacity: unset;
                display: flex;
            }
        }
        
    }
}