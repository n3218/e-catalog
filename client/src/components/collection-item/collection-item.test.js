import { shallow, mount, render } from "enzyme"
import React from "react"
import { CollectionItem } from "./collection-item.component"

const mockItem = {
  item: {
    imageUrl: "http://imageurl.html",
    price: "$20",
    name: "name of Product",
    quantity: "2"
  }
}

const mockAddItem = () => {}

const wrapper = shallow(<CollectionItem item={mockItem} addItem={mockAddItem} />)

it("expect to render CollectionItem component", () => {
  expect(wrapper).toMatchSnapshot()
})

it("Correctly increments the counter", () => {
  wrapper.find("CustomButton").simulate("click")
})
