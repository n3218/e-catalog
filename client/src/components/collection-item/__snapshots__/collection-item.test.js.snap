// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render CollectionItem component 1`] = `ShallowWrapper {}`;
